matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/summary/RoomSummaryUpdater.kt

```diff
@@ -180,8 +182,14 @@ internal class RoomSummaryUpdater @Inject constructor(
             roomSummaryEntity.otherMemberIds.clear()
             roomSummaryEntity.otherMemberIds.addAll(otherRoomMembers)
             if (roomSummaryEntity.isEncrypted && otherRoomMembers.isNotEmpty()) {
-                // mmm maybe we could only refresh shield instead of checking trust also?
-                crossSigningService.onUsersDeviceUpdate(otherRoomMembers)
+                if (aggregator == null) {
```

Comment:
I struggle to understand why using the `aggregator` changes something here. Seeing the implementation of `onUsersDeviceUpdate`, I see we only schedule a worker we should not take so much time, right?

Reply:
In the current code, `crossSigningService.onUsersDeviceUpdate(otherRoomMembers)` is called once per room.
With the new code this is called once per sync response (which can contains many rooms)

Reply:
Okay thanks for explaining.